src/main/java/org/jabref/gui/util/UiThreadList.java

```diff
@@ -1,18 +1,38 @@
 package org.jabref.gui.util;
 
+import java.util.concurrent.CountDownLatch;
+
 import javafx.application.Platform;
 import javafx.collections.ListChangeListener;
 import javafx.collections.ObservableList;
 import javafx.collections.transformation.TransformationList;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 class UiThreadList<T> extends TransformationList<T, T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UiThreadList.class);
     public UiThreadList(ObservableList<? extends T> source) {
         super(source);
     }
 
     @Override
     protected void sourceChanged(ListChangeListener.Change<? extends T> change) {
-        Platform.runLater(() -> fireChange(change));
+        if (Platform.isFxApplicationThread()) {
```

Comment:
Oh, wow. This reads like an ugly hack. Good, that this works...

Would be nice if some comments were added:

- Why can it happen that this method is called inside and outside the FxApplication thread?
- Why can the `fireChange` method be run directly in the case of inside the FxApplication?
- Why is `CountDownLatch` used and not something more different (it is used as Boolean flag)?